metadata/src/lib.rs

```diff
@@ -61,6 +65,63 @@ pub struct Metadata {
     custom: frame_metadata::v15::CustomMetadata<PortableForm>,
 }
 
+// Since we've abstracted away from frame-metadatas, we impl this on our custom Metadata
+// so that it can be used by `frame-decode` to obtain the relevant extrinsic info.
+impl frame_decode::extrinsics::ExtrinsicTypeInfo for Metadata {
```

Comment:
We wouldn't normally need this if we used frame_metadata's `RuntimeMetadataV15` or whatever directly, but since we abstracted it into our own Metadata crate so as to not care about the underlying version, we need to be able to get the info we need from this via the impl here.